lang/gcc13/Portfile

```diff
@@ -0,0 +1,561 @@
+    }
+
+}
+
+set libcxx_incpath ${prefix}/libexec/${name}/libc++/include/c++/v1
+
+subport ${libcxxname} {
+    PortGroup compilers     1.0
+
+    compilers.choose        cxx
+    compilers.setup         -gcc -fortran -clangdevel -clang34 -clang35 -clang36 \
+                            -clang37 -clang50 -clang60 -clang70 -clang80 -clang90
+    if { ![clang_variant_isset] } {
+        if { ${os.platform} eq "darwin" && ${os.major} < 11 } {
+            default_variants-append +clang11
+        } else {
+            default_variants-append +clang16
+        }
+    }
+    revision                [ expr ${revision} + 0 ]
+    description             libc++ header implementation to be used by ${name}
+    long_description        {*}${description}
+    homepage                https://llvm.org/
+    license                 NCSA
+    depends_build
+    depends_extract
+    depends_run
+    depends_lib
+    distfiles
+    patchfiles
+    use_configure           no
+    supported_archs         noarch
+    # Find clang/llvm version to use from active variant
+    proc getClangVersion {} {
+        if { [regexp {clang(.*)} [clang_variant_name] -> clang_v] } {
+            return ${clang_v}
+        }
+        return ""
+    }
+    set mp_clang_ver [getClangVersion]
+    build {
+        # Copy headers from clang-N during build
+        file mkdir ${worksrcpath}/headers
+        file copy ${prefix}/libexec/llvm-${mp_clang_ver}/include/c++/v1 ${worksrcpath}/headers/
+    }
+    destroot {
+        set base_dir [file dirname ${libcxx_incpath}]
+        file mkdir ${destroot}${base_dir}
+        file copy ${worksrcpath}/headers/v1 ${destroot}${base_dir}
+    }
+}
+
+if { ${subport} ne ${libcxxname} } {
+
+    variant stdlib_flag description {Enable stdlib command line flag to select c++ runtime} {
+        # Enables support for specifying the c++ runtime via `-stdlib=` in a similar
+        # way to clang. For more details see the later comments in
+        #   https://www.mail-archive.com/gcc-patches@gcc.gnu.org/msg257385.html
+        # Note : This 'bakes' the libc++ include directory into gcc,
+        # which is then used as the default search location when `-stdlib=libc++`
+        # is given. Therefore to have consistency across various OS versions, and to enable
+        # modern c++ standards, use a recent macports clang port to provide this.
+        if {${subport} eq ${name}} {
+            depends_run-append port:${libcxxname}
+        }
+        configure.args-append --with-gxx-libcxx-include-dir="${libcxx_incpath}"
+    }
+
+    # libcxx is unavailable on PPC
+    if {${build_arch} ni [list ppc ppc64]} {
+        default_variants-append +stdlib_flag
+    }
+
+}
+
+# https://trac.macports.org/ticket/29067
+# https://trac.macports.org/ticket/29104
+# https://trac.macports.org/ticket/47996
+# https://trac.macports.org/ticket/58493
+compiler.blacklist-append {clang < 800} gcc-4.0 *gcc-4.2 {llvm-gcc-4.2 < 2336.1} {macports-clang-3.[4-7]}
```

Comment:
Perhaps, `*gcc-4.0`, since there is `apple-gcc40` port, we do not want to use it for sure.

However, it makes better sense to drop the ban of old GCCs and instead set `cxx_standard 2011`, since GCC 11+ does require C++11. That will take care of old GCC, and also will unbreak building it with `gcc10-bootstrap` (for w/e reason it does not work with existing blacklist – at least it did not work for gcc12).

Reply:
For various reasons we tend to avoid the use of the compiler blacklisting here and prefer to hand tune things.

Reply:
I want to see how the port builds with the current settings, on all the OSes we support, before making any changes to the compiler selection.

Reply:
Well, it currently uses `compiler.blacklist-append`, and the last time I checked with `gcc12` it did not work well, leading to a circular dependency when trying to build with `gcc10-bootstrap` (should be a bug somewhere in Macports, but I do not know what causes that). This may or may not apply to this portfile – I need to check it.

Reply:
Sorry, I misspoke...

We currently avoid setting the c++ standard, not the blacklist.

I am not saying we perhaps could not consider changing things, but before doing that I want to see how things stand as they are right now.